components/camel-kafka/src/main/java/org/apache/camel/component/kafka/KafkaFetchRecords.java

```diff
@@ -242,6 +242,8 @@ private void handleAccordingToStrategy(long partitionLastOffset, Exception e) {
     private void safeUnsubscribe() {
         try {
             consumer.unsubscribe();
+        } catch (IllegalStateException e) {
+            LOG.warn("The consumer is likely already closed. Skipping the unsubscription");
```

Comment:
I think we can use some of the same information from below about which topicName etc it was

Reply:
Thanks, I think that's a good idea. I am on my way back home, so I take a look tomorrow to see what we can print.

Reply:
This should be fine now. I also logged CAMEL-17509 for some other things I noticed based on your suggestion.